### ISSUE TITLE
[Question] waitFor state behavior different between firefox and chromium/webkit

### ISSUE THREAD

**Issue by kamelbou** [microsoft/playwright#13351]
Hello !

**Playwright version: 1.20.2**
**OS: Mac OS**

I have a different behavior between firefox and chromium/webkit for waitFor state.

***
Firefox
```js
const frmElement = await page.frameLocator('.ec-app-host-item');
await frmElement.locator('html').waitFor({state: 'visible'});
```
***

Chromium/webkit
```js
const frmElement = await page.frameLocator('.ec-app-host-item');
await frmElement.locator('html').waitFor({state: 'hidden'});
```

For chromium/webkit, if I put visible I have this issue:

```
locator.waitFor: Timeout 30000ms exceeded.
       =========================== logs ===========================
       waiting for selector ".ec-app-host-item >> control=enter-frame >> html" to be visible
         waiting for frame ".ec-app-host-item"
         selector resolved to visible <iframe class="height-full width-full ec-app-host-item" …></iframe>
         selector resolved to hidden <html>…</html>
```

Is that something related to the structure of DOM ?

<img width="117" alt="Capture d’écran 2022-04-06 à 17 05 13" src="https://user-images.githubusercontent.com/37103602/162006439-5a97af76-6db1-44d0-93a1-b062cdde6bf4.png">

Thank you

**Comment by mxschmitt** [comment]
What exactly is the difference? Since the logs of the first and second invocation are different API calls its hard to follow. Having a repro would it would help a lot!

**Comment by kamelbou** [comment]
> What exactly is the difference? Since the logs of the first and second invocation are different API calls its hard to follow. Having a repro would it would help a lot!

In the first case, with Firefox, I put state to visible and the test is Pass. If I keep visible state for chromium and webkit, the test is Fail.

**Comment by garciallorente-fer** [comment]
I have a very similar problem, I have a test case in which I work with elements inside frames, and the test works in Chromium but not in Firefox or Webkit, because the frame is HIDDEN that makes it fail in Firefox and Webkit, but works fine in Chromium.

const frameLocator = this.page.frameLocator('iframe');
const locator = frameLocator.locator('[name=textinput]');
console.log(await locator.isHidden()); // True ALWAYS
await locator.waitFor({ state: 'attached' }); // Fails ONLY in FIREFOX & WEBKIT
await locator.fill('some text'); // Fails ONLY in FIREFOX & WEBKIT

```
waiting for frame "iframe"
      selector resolved to hidden <iframe scrolling="no" frameborder="0" allowtransparency…></iframe>
    waiting for selector "iframe >> control=enter-frame >> [name=textinput]"
```

**Comment by pavelfeldman** [comment]
What do you mean when you say that the frame is HIDDEN? Could you share a snippet?

**Comment by pavelfeldman** [comment]
Was this resolved?

**Comment by pavelfeldman** [comment]
> I dont know why in Chromium this same code works but fails in Firefox.

Nothing unusual here, that's why we have cross-browser testing. So according to the logs, it detects iframe, but times out waiting for the element. The logs seem to be truncated though, it should tell us it now waits for selector in that frame. Could you record a trace (`trace: on` in config)?. Btw, you should only need one line `await locator.fill('some text')`, no need to wait for anything, Playwright will do it all.

**Comment by pavelfeldman** [comment]
We need more information to act on this report. Please file a new one and link to this issue when you get back to it!

**Comment by garciallorente-fer** [comment]
> > I dont know why in Chromium this same code works but fails in Firefox.
> 
> Nothing unusual here, that's why we have cross-browser testing. So according to the logs, it detects iframe, but times out waiting for the element. The logs seem to be truncated though, it should tell us it now waits for selector in that frame. Could you record a trace (`trace: on` in config)?. Btw, you should only need one line `await locator.fill('some text')`, no need to wait for anything, Playwright will do it all.

Thanks for the response @pavelfeldman  With the trace on this is the information I get from it: 
```
Action was interrupted
locator.fill
TIME
wall time: 04/05/2022, 17:43:19
duration: Timed Out
PARAMETERS
selector: "[id="card-number-element"] iframe >> control=enter-frame >> input[type=text][name=cardnumber]"
strict: true
value: "4000 0000 0000 0002"
LOG
waiting for frame "[id="card-number-element"] iframe"
selector resolved to visible <iframe scrolling="no" frameborder="0" allowtransparency…></iframe>
waiting for selector "[id="card-number-element"] iframe >> control=enter-frame >> input[type=text][name=cardnumber]"
```

Why are the logs truncated though? if they should tell us it now waits for selector in that frame.
Why it times out waiting for the element in Firefox?
